### DISCUSSION TITLE
Should lsblk display fuse-mounted devices?

### DISCUSSION THREAD

**Discussion by ahmogit** [util-linux/util-linux#1781] [Ideas]
@karelzak Would it make sense for lsblk to have the capability of listing fuse-mounted devices?

If so, I will be happy to file a feature request issue for it, but figured better to ask first.  My familiarity with the semantics of "block device" is too weak to appreciate whether a fuse-mounted device should even legitimately be considered as such. Maybe it doesn't even make sense.

**Comment by karelzak** [comment]
Not sure if I understand the question. What do you mean by `fuse-mounted devices`? Do you have any example?

**Comment by ahmogit** [comment]
Example: 

Suppose `/etc/fstab' contains this line:

     ifuse         /mnt/whatever             fuse.ifuse noauto,user,nosuid,noexec

and is then user-mounted via

     $ ifuse /mnt/whatever


That filesystem can then be interrogated in various ways, e.g. via (e.g.) `df(1)`,  `findmnt(1)`, etc.:
```
     $ df /mnt/whatever
     Filesystem          1K-blocks     Used     Available   Use%   Mounted on
     ifuse               123456789     123456   12345678    12%    /mnt/whatever

    $ findmnt /mnt/whatever
     TARGET            SOURCE    FSTYPE        OPTIONS
     /mnt/whatever     ifuse     fuse.ifuse    rw,nosuid,nodev,relatime,user_id=xxx,group_id=yyy
```

Thus, w.r.t. these tools, `/mnt/whatever` "sorta looks like" a block device.

So the question is: **Would it make sense for `lsblk` to also have the ability to display whatever info that it can about `/mnt/whatever`?**

Currently, it just reports `/mnt/whatever is not a block device`.

**Comment by karelzak** [comment]
I don't think in this case it's a good idea. 

First, in your case /mnt/whatever is a mountpoint, the place where the filesystem is attached to the tree (VFS). It's a directory, definitely not a  block device.

Generally speaking, the filesystem, from the kernel point of view, is a standalone object which is attached (mounted) to VFS. It's filesystem-specific things if the filesystem uses any block device, or network, or it generates data on the fly. So, the connection between FS and the block device could be pretty vague. The block device is always something you can see in /sys/block, never anything else. And don't forget that kernel provides many attributes for block devices, nothing is provided for your /mnt/whatever -- in other words, lsblk will have nothing to display.

For classic filesystems (extN, XFS, ...), where is 1:1 mapping between FS and block device and where we have a proper entry with /dev/something source in /proc/self/mountinfo it makes sense for lsblk to display the mountpoint. For complex cases like btrfs it's already tricky, and for other filesystems, it's impossible because there is no block device (NFS, CIFS, many fsuse).

I understand you want to see all in out ls-like output :-), but in this case, you need to use two tools, one for filesystems (e.g., findmnt) and the second for block devices (lsblk).
